tests/test_search.py

```diff
@@ -346,3 +349,20 @@ def assert_is_sorted(item, path: str):
             assert item == sorted(item), f'{err_path} is not sorted'
         for i, child in enumerate(item):
             assert_is_sorted(child, f'{path}[{i}]')
+
+
+@pytest.mark.parametrize('directory', (
+    directory for directory in (TESTS_ROOT / 'js' / 'roots').iterdir()
+))
+def test_check_js_search_indexes(make_app, sphinx_test_tempdir, directory):
+    app = make_app('html', srcdir=directory, builddir=sphinx_test_tempdir / directory.name)
+    app.build()
+
+    fresh_searchindex = (app.outdir / 'searchindex.js')
+    fresh_hash = sha256(fresh_searchindex.read_bytes()).digest()
```

Comment:
It's a bit unclear to me why you're taking a hash of the file rather than just comparing the contents directly? Feels like that would be just as fast and probably simpler given the sizes of our files.

Reply:
Initially I felt some concern about this code perhaps being reused for larger files -- to the extent that entire-file comparison might not be ideal -- however, it is true that the search indexes are small, and so the simplicity benefit of comparing the full content does seem worthwhile.